core_types/src/eav.rs

```diff
@@ -19,13 +19,36 @@ use std::{
 };
 
 use regex::RegexBuilder;
-use std::fmt::Debug;
+use std::fmt::{self, Debug};
 
 /// Address of AddressableContent representing the EAV entity
 pub type Entity = Address;
 
 /// Using String for EAV attributes (not e.g. an enum) keeps it simple and open
-pub type Attribute = String;
+#[derive(PartialEq, Eq, PartialOrd, Hash, Clone, Debug, Serialize, Deserialize, DefaultJson)]
+pub enum Attribute {
+    #[serde(rename = "crud-status")]
+    CrudStatus,
+
+    #[serde(rename = "crud-link")]
+    CrudLink,
+
+    #[serde(rename = "link")]
+    Link,
+
+    LinkTag(String),
+}
+
+impl fmt::Display for Attribute {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
```

Comment:
Here we have some dashes and some underscores, anything we can do about that?

Reply:
This is just what we've always had, but now would be a good time to standardize the names since they're all in one place now.

Reply:
I agree and we should also choose an automation method for this

Reply:
@sphinxc0re how?

Reply:
I thought there might be a derive macro for Display and FromStr somewhere

Reply:
https://crates.io/crates/strum

Reply:
relevant context, fyi only - https://github.com/holochain/holochain-rust/pull/750/files - we previously broke windows via A names that weren't file system friendly